gm4_blossoming_pots/data/gm4_blossoming_pots/function/flower/set_displays.mcfunction

```
@@ -1,17 +1,11 @@
 # Main Loop for setting displays to desired stage
 # @s = temp entity inside flower pot
 # at @s align xyz positioned ~.5 ~.5 ~.5
-# with {rotation, display_number, data}
-# run from flower/as_marker and self
+# with {data}
+# run from flower/prepare_data and self
 
-# summon if needed
-$execute unless entity @e[type=minecraft:block_display,distance=..0.1,tag=gm4_blossoming_pots.display.flower_pot.$(display_number),limit=1] \
-    run summon minecraft:block_display ~ ~ ~ {view_range:0.7935f,Tags:["gm4_blossoming_pots.display.flower_pot","gm4_blossoming_pots.display.flower_pot.$(display_number)","smithed.entity"],Rotation:[$(rotation)f,0f]}
-# merge data and store the success
-$execute store success score $merge_success gm4_blossoming_pots.misc \
-    run data merge entity @e[type=minecraft:block_display,distance=..0.1,tag=gm4_blossoming_pots.display.flower_pot.$(display_number),limit=1] $(data)
-# if a single merge success set sound to play
-execute if score $merge_success gm4_blossoming_pots.misc matches 1 run scoreboard players set $play_sound gm4_blossoming_pots.misc 1
+# create display
+$summon minecraft:block_display ~ ~ ~ $(data)
```

I guess this technically works. I usually prefer not to use macros unless necessary, tho. You could use `execute summon` here instead.

---

Changing to `execute summon block_display run data modify entity @s {} merge from storage gm4_blossoming_pots:flower_pots temp` caused a silent error with `block_state` being set to air and not carrying the transformation data.